src/SAML2/SOAP.php

```diff
@@ -16,9 +19,15 @@ class SAML2_SOAP extends SAML2_Binding
      */
     public function send(SAML2_Message $message)
     {
+        assert('!empty($this->AssertionConsumerServiceURL)');
         header('Content-Type: text/xml', TRUE);
         $outputFromIdp = '<?xml version="1.0" encoding="UTF-8"?>';
         $outputFromIdp .= '<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/">';
+        $outputFromIdp .= '<SOAP-ENV:Header>';
+        $outputFromIdp .= '<ecp:Response xmlns:ecp="urn:oasis:names:tc:SAML:2.0:profiles:SSO:ecp" ';
+        $outputFromIdp .= '  SOAP-ENV:mustUnderstand="1" SOAP-ENV:actor="http://schemas.xmlsoap.org/soap/actor/next" ';
+        $outputFromIdp .= '  AssertionConsumerServiceURL="' . $this->AssertionConsumerServiceURL . '"/>';
```

Comment:
IMHO, this method should differentiate what kind of `SAML2_Message` is getting (*instanceof*), and act accordingly, instead of using an extra property that's not used and that leads to inconsistencies. Then, if this is a SAML response, the method should build this SOAP header with the `ecp:Response` inside, and add an `AssertionConsumerServiceURL` attribute, using the value located in `$this->destination`.